Lib/javascript/napi/javascriptcode.swg

```diff
@@ -758,12 +758,10 @@ void $jsmangledname_inst::GetMembers(
   // jsnapi_registerclass
   Napi::Function $jsmangledname_ctor = $jsmangledname_inst::GetClass(env);
   $jsparent.Set("$jsname", $jsmangledname_ctor);
-  if (SWIGTYPE_$jsmangledtype->clientdata == nullptr) {
-    SWIGTYPE_$jsmangledtype->clientdata = new size_t($jsclassidx);
-  }
-  Napi::FunctionReference *$jsmangledname_ctor_ref = new Napi::FunctionReference();
-  *$jsmangledname_ctor_ref = Napi::Persistent($jsmangledname_ctor);
-  env.GetInstanceData<EnvInstanceData>()->ctor[$jsclassidx] = $jsmangledname_ctor_ref;
+  // Almost, but not quite, entirely unlike random bias of 42 to disambiguate
```

Comment:
Why 42. Surely initialise to something sensible like zero.

Reply:
I don't really know. It does solve some of the leaks, but it abuses a pointer type to store a number.

Alas, it does not really solve the module leaking memory when it is unloaded. I am using a deprecated form of the call used to to assign getters and setters because the recommended form does not support CRTP classes. I spent some time trying to find a solution and I couldn't do it. The recommended form expects the getter and the setter function to be passed as template arguments.

I can only guess that this is the reason that the Node.js team has left this deprecated form as it is.

Alas, clean unloading of binary modules in Node.js is difficult and most modules do not support it. It is a rare situation - it requires that the module is loaded by a `worker_thread` and then the `worker_thread` quits. In this case Node.js will unload the shared library. Me, personally, I recommend that all my users load all my binary modules from the main thread before referencing them in a `worker_thread`.

Anyway, I do not think that this is the most pressing issue at the moment - but I do not see any good reason neither to merge it or to not merge it.